tools/transform_quic_trace.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A helper tool to perform various transformations on traces, notably
+// protobuf-to-JSON conversion and trace truncation.
+
+#include <iostream>
+#include <istream>
+
+#include "gflags/gflags.h"
+#include "google/protobuf/util/json_util.h"
+#include "lib/quic_trace.pb.h"
+
+DEFINE_bool(whitespace, true, "Add whitespace to the JSON output");
+DEFINE_string(input_format, "protobuf", "Can be 'protobuf' or 'json'.");
+DEFINE_string(output_format, "protobuf", "Can be 'protobuf' or 'json'.");
+DEFINE_int64(truncate_after_time_us,
```

Comment:
You can remove the end, but not the beginning of a trace?  Any reason?

Reply:
You could remove the beginning, but you'd have to recompute the timestamps, which is something I didn't want to write yet.